plasmapy/diagnostics/thomson.py

```diff
@@ -85,24 +340,24 @@ def spectral_density(
 
     n : `~astropy.units.Quantity`
         Mean (0th order) density of all plasma components combined.
-        (convertible to cm\ :sup:`-3`\ .)
+        (convertible to cm\ :sup:`-3`)
 
     Te : `~astropy.units.Quantity`, shape (Ne, )
         Temperature of each electron component. Shape (Ne, ) must be equal to the
-        number of electron components Ne. (in K or convertible to eV)
+        number of electron populations Ne. (in K or convertible to eV)
 
     Ti : `~astropy.units.Quantity`, shape (Ni, )
         Temperature of each ion component. Shape (Ni, ) must be equal to the
-        number of ion components Ni. (in K or convertible to eV)
+        number of ion populations Ni. (in K or convertible to eV)
 
     efract : array_like, shape (Ne, ), optional
         An array-like object where each element represents the fraction (or ratio)
-        of the electron component number density to the total electron number density.
+        of the electron population number density to the total electron number density.
         Must sum to 1.0. Default is a single electron component.
 
     ifract : array_like, shape (Ni, ), optional
         An array-like object where each element represents the fraction (or ratio)
-        of the ion component number density to the total ion number density.
+        of the ion population number density to the total ion number density.
         Must sum to 1.0. Default is a single ion species.
```

Comment:
This is a bit miss leading.  I think it's too easy for a user to mistakenly think you're talking about `n` here when you're actually talking about the total ion number density `np.sum(ion_z * ni)`.  That is, `ifract = ni / np.sum(ion_z * ni)`.

---
Now, if I'm understanding this correctly, then I agree `np.sum(ifract) == 1`, but I think your `ni` calculation is wrong.  Instead of `ni = ifract * n / zbar` it should be `ni = (ifract / ion_z) * n`.

![image](https://user-images.githubusercontent.com/29869348/163513245-1e6cc5f7-75fa-4970-9510-f3d154bcb42c.png)

Reply:
So I chose this definition because it's how it's done in the code provided in the Sheffield book, but I've also found it unintuitive so here's some plugged-in numbers. 

n=10
F = [0.7, 0.3] 
Z = [3, 2] 
Zbar = 2.7 

n_i
implemented: [2.593,  1.111]
yours:  [2.333,  1.500]

Charge density = \sum n_i * Z_i
implemented:  10 = n
yours: 10 = n

So both ways maintain charge neutrality, and I'd argue are both are reasonable definitions. 

But the implemented way has the additional property that 
n_i = F_i * \sum n_i 

Total ion density (\sum n_i) is 
implemented: 3.704   (equal to 10/Zbar)
yours: 3.833

F_i * \sum n_i 
implemented: [2.593, 1.111]  = n_i 
yours: [2.683, 1.149] != n_i 

Meaning that the F_i as 'fractions' of the total ion density only makes sense with this definition. So, I think the statement in the docstring is actually true, although I'd agree it's definitely not obvious why it is? I've added more explicit clarification about the definition of n_i in the docstring so people know what definition is being used.

Reply:
Ok, the part I was missing was total ion-densities being equal to total electron density divided by zbar, which after some quick math can be validated.  I agree with the current implementation now.

![image](https://user-images.githubusercontent.com/29869348/163844777-a098ed35-b077-423f-a367-e0abb63426c9.png)